pkg/statistics/handle/handletest/statstest/stats_test.go

```diff
@@ -311,6 +311,393 @@ num: 1 lower_bound: 7 upper_bound: 7 repeats: 1 ndv: 0`, tbl.Meta().ID)
+		require.False(t, col.IsFullLoad())
+		require.Equal(t, int64(0), col.NDV)
+		require.Equal(t, int64(0), col.NullCount)
+		require.Equal(t, float64(0), col.TotalRowCount())
+		require.Nil(t, col.TopN)
+		require.Equal(t, 0, col.Histogram.Len())
+		return false
+	})
+
+	// Another table with predicate columns
+	tk.MustExec("create table t2(a int, b int, c int, primary key(a), key idx(b))")
+	tk.MustExec("insert into t2 values (1,1,1),(2,2,2),(3,3,3),(4,4,4),(5,5,5),(6,7,8)")
+	h = dom.StatsHandle()
+	is = dom.InfoSchema()
+	// Handle DDL event to init the stats meta and histogram meta.
+	err = statstestutil.HandleNextDDLEventWithTxn(h)
+	require.NoError(t, err)
+	tk.MustExec("analyze table t2 with 2 topn, 2 buckets")
+	tbl2, err := is.TableByName(context.Background(), ast.NewCIStr("test"), ast.NewCIStr("t2"))
+	require.NoError(t, err)
+
+	h.Clear()
+	require.NoError(t, h.InitStats(context.Background(), is))
+	tableStats2 := h.GetPhysicalTableStats(tbl2.Meta().ID, tbl2.Meta())
+	require.True(t, tableStats2.IsAnalyzed())
+	// Check index stats
+	require.Equal(t, 1, tableStats2.IdxNum())
+	idx2 := tableStats2.GetIdx(tbl2.Meta().Indices[0].ID)
+	require.NotNil(t, idx2)
+	require.True(t, tableStats2.ColAndIdxExistenceMap.Has(idx2.ID, true))
+	require.True(t, tableStats2.ColAndIdxExistenceMap.HasAnalyzed(idx2.ID, true))
+	require.True(t, idx2.IsStatsInitialized())
+	require.False(t, idx2.IsAllEvicted())
+	require.True(t, idx2.IsFullLoad())
+	require.Equal(t, uint64(2), idx2.TopN.TotalCount())
+	require.Equal(t, float64(6), idx2.TotalRowCount())
+	require.Equal(t, 2, idx2.Histogram.Len())
+	// Check column stats
+	require.Equal(t, int(3), tableStats2.ColNum())
+	// For column c, it is not in the predicate columns, so its stats has not been collected.
+	tableStats2.ForEachColumnImmutable(func(_ int64, col *statistics.Column) bool {
+		if col.Info.Name.L == "c" {
+			require.True(t, tableStats2.ColAndIdxExistenceMap.Has(col.ID, false))
+			require.False(t, tableStats2.ColAndIdxExistenceMap.HasAnalyzed(col.ID, false))
+			require.False(t, col.IsStatsInitialized())
+			require.False(t, col.IsAllEvicted())
+			require.False(t, col.IsFullLoad())
+			require.Equal(t, int64(0), col.NDV)
+			require.Equal(t, int64(0), col.NullCount)
+			require.Equal(t, float64(0), col.TotalRowCount())
+			require.Nil(t, col.TopN)
+			require.Equal(t, 0, col.Histogram.Len())
+			return false
+		}
+		require.True(t, tableStats2.ColAndIdxExistenceMap.Has(col.ID, false))
+		require.True(t, tableStats2.ColAndIdxExistenceMap.HasAnalyzed(col.ID, false))
+		require.True(t, col.IsStatsInitialized())
+		require.True(t, col.IsAllEvicted())
+		require.False(t, col.IsFullLoad())
+		require.Equal(t, int64(6), col.NDV)
+		require.Equal(t, int64(0), col.NullCount)
+		require.Equal(t, float64(0), col.TotalRowCount())
+		require.Nil(t, col.TopN)
+		require.Equal(t, 0, col.Histogram.Len())
+		return false
+	})
+}
+
+func TestInitStatsForPartitionedTable(t *testing.T) {
```

Comment:
I know this test is very similar to the non-partitioned one, but I prefer not to share code between them. For these tests, the status checks are very detailed, and moving them into a separate function would significantly reduce readability when reviewing the tests.